logger.go

```diff
@@ -148,21 +162,25 @@ func parseFormat(format string) (msgfmt, timefmt string) {
 // (now used only as time format)
 func ph2verb(ph string) (verb string, arg string) {
 	n := len(ph)
-	if n < 4 { return ``, `` }
-	if ph[0] != '%' || ph[1] != '{' || ph[n-1] != '}' { return ``, `` }
+	if n < 4 {
+		return ``, ``
+	}
+	if ph[0] != '%' || ph[1] != '{' || ph[n-1] != '}' {
+		return ``, ``
+	}
 	idx := strings.IndexRune(ph, ':')
 	if idx == -1 {
 		return phfs[ph], ``
 	}
-	verb = phfs[ ph[:idx] + "}" ]
-	arg = ph[idx+1:n-1]
+	verb = phfs[ph[:idx]+"}"]
+	arg = ph[idx+1 : n-1]
 	return
 }
 
 // Returns an instance of worker class, prefix is the string attached to every log,
 // flag determine the log params, color parameters verifies whether we need colored outputs or not
-func NewWorker(prefix string, flag int, color int, out io.Writer) *Worker {
-	return &Worker{Minion: log.New(out, prefix, flag), Color: color, format: defFmt, timeFormat: defTimeFmt}
+func NewWorker(prefix string, flag int, color int, out io.Writer, level LogLevel) *Worker {
```

Comment:
This needs to be backwards compatible, otherwise it will break a lot of codebases.